styleguide/newline.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python3
+
+import os
+import re
+import sys
+
+sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
+import wpi
+
+# Files and directories which should be included in or excluded from processing
+regexExclude = re.compile(wpi.regexExclude())
+
+# Handle running in either the root or styleguide directories
+configPath = ""
+if os.getcwd().rpartition(os.sep)[2] == "styleguide":
+    configPath = ".."
+else:
+    configPath = "."
+
+# Recursively create list of files in given directory
+files = [os.path.join(dp, f) for dp, dn, fn in
+         os.walk(os.path.expanduser(configPath)) for f in fn]
+
+# Apply regex filters to list
+files = [f for f in files if not regexExclude.search(f)]
+
+if not files:
+    print("Error: no files found to format", file=sys.stderr)
+    sys.exit(1)
+
+for name in files:
+    # List names of files as they are processed if verbose flag was given
+    if len(sys.argv) > 1 and sys.argv[1] == "-v":
+        print("Processing", name,)
+
+    newlines = 0
+
+    # Remove all EOF newlines, then append one
+    with open(name, "r+", encoding = "ISO-8859-1") as file:
```

Comment:
This happened when I last used UTF-8:
`
[tav@myriad allwpilib]$ ./license-update.py -v
Processing ./hal/include/Log.h
Processing ./hal/include/HAL/Port.h
Processing ./hal/include/HAL/Accelerometer.h
Processing ./hal/include/HAL/AnalogAccumulator.h
Processing ./hal/include/HAL/AnalogInput.h
Processing ./hal/include/HAL/AnalogOutput.h
Processing ./hal/include/HAL/AnalogTrigger.h
Processing ./hal/include/HAL/CAN.h
Processing ./hal/include/HAL/Compressor.h
Processing ./hal/include/HAL/Counter.h
Processing ./hal/include/HAL/DIO.h
Processing ./hal/include/HAL/Encoder.h
Processing ./hal/include/HAL/Errors.h
Processing ./hal/include/HAL/HAL.h
Processing ./hal/include/HAL/I2C.h
Processing ./hal/include/HAL/Interrupts.h
Processing ./hal/include/HAL/Notifier.h
Processing ./hal/include/HAL/PDP.h
Processing ./hal/include/HAL/PWM.h
Processing ./hal/include/HAL/Power.h
Processing ./hal/include/HAL/Relay.h
Processing ./hal/include/HAL/SPI.h
Processing ./hal/include/HAL/Semaphore.h
Processing ./hal/include/HAL/SerialPort.h
Processing ./hal/include/HAL/Solenoid.h
Processing ./hal/include/HAL/Task.h
Processing ./hal/include/HAL/cpp/priority_mutex.h
Processing ./hal/include/HAL/cpp/Resource.h
Processing ./hal/include/HAL/cpp/Semaphore.h
Processing ./hal/include/HAL/cpp/priority_condition_variable.h
Processing ./hal/lib/athena/ChipObject.h
Processing ./hal/lib/athena/Semaphore.cpp
Processing ./hal/lib/athena/Accelerometer.cpp
Processing ./hal/lib/athena/AnalogAccumulator.cpp
Processing ./hal/lib/athena/AnalogInput.cpp
Processing ./hal/lib/athena/AnalogInternal.cpp
Processing ./hal/lib/athena/AnalogInternal.h
Processing ./hal/lib/athena/AnalogOutput.cpp
Processing ./hal/lib/athena/AnalogTrigger.cpp
Processing ./hal/lib/athena/Compressor.cpp
Processing ./hal/lib/athena/Counter.cpp
Processing ./hal/lib/athena/DIO.cpp
Processing ./hal/lib/athena/DigitalInternal.cpp
Processing ./hal/lib/athena/DigitalInternal.h
Processing ./hal/lib/athena/Encoder.cpp
Processing ./hal/lib/athena/HALAthena.cpp
Processing ./hal/lib/athena/I2C.cpp
Processing ./hal/lib/athena/Interrupts.cpp
Processing ./hal/lib/athena/Notifier.cpp
Processing ./hal/lib/athena/PDP.cpp
Processing ./hal/lib/athena/PWM.cpp
Processing ./hal/lib/athena/Power.cpp
Processing ./hal/lib/athena/Relay.cpp
Processing ./hal/lib/athena/SPI.cpp
Processing ./hal/lib/athena/SerialPort.cpp
Processing ./hal/lib/athena/Solenoid.cpp
Processing ./hal/lib/athena/Task.cpp
Processing ./hal/lib/athena/cpp/Semaphore.cpp
Processing ./hal/lib/athena/cpp/priority_mutex.cpp
Processing ./hal/lib/athena/cpp/Resource.cpp
Processing ./hal/lib/shared/HAL.cpp
Processing ./hal/lib/sim/HALDesktop.cpp
Processing ./simulation/JavaGazebo/src/main/java/org/gazebosim/transport/Connection.java
Processing ./simulation/JavaGazebo/src/main/java/org/gazebosim/transport/Msgs.java
Processing ./simulation/JavaGazebo/src/main/java/org/gazebosim/transport/Node.java
Processing ./simulation/JavaGazebo/src/main/java/org/gazebosim/transport/Publisher.java
Processing ./simulation/JavaGazebo/src/main/java/org/gazebosim/transport/PublisherRecord.java
Processing ./simulation/JavaGazebo/src/main/java/org/gazebosim/transport/RemotePublisherRecord.java
Processing ./simulation/JavaGazebo/src/main/java/org/gazebosim/transport/ServerCallback.java
Processing ./simulation/JavaGazebo/src/main/java/org/gazebosim/transport/Subscriber.java
Processing ./simulation/JavaGazebo/src/main/java/org/gazebosim/transport/SubscriberCallback.java
Processing ./simulation/SimDS/src/main/java/edu/wpi/first/wpilibj/simulation/ds/DS.java
Processing ./simulation/SimDS/src/main/java/edu/wpi/first/wpilibj/simulation/ds/EnableAction.java
Processing ./simulation/SimDS/src/main/java/edu/wpi/first/wpilibj/simulation/ds/FakeJoystick.java
Processing ./simulation/SimDS/src/main/java/edu/wpi/first/wpilibj/simulation/ds/ISimJoystick.java
Processing ./simulation/SimDS/src/main/java/edu/wpi/first/wpilibj/simulation/ds/JoystickList.java
Processing ./simulation/SimDS/src/main/java/edu/wpi/first/wpilibj/simulation/ds/JoystickProvider.java
Processing ./simulation/SimDS/src/main/java/edu/wpi/first/wpilibj/simulation/ds/Main.java
Processing ./simulation/SimDS/src/main/java/edu/wpi/first/wpilibj/simulation/ds/ModeAction.java
Processing ./simulation/SimDS/src/main/java/edu/wpi/first/wpilibj/simulation/ds/RefreshAction.java
Processing ./simulation/SimDS/src/main/java/edu/wpi/first/wpilibj/simulation/ds/SimJoystick.java
Processing ./simulation/frc_gazebo_plugins/clock/src/clock.cpp
Processing ./simulation/frc_gazebo_plugins/clock/src/clock.h
Processing ./simulation/frc_gazebo_plugins/dc_motor/src/dc_motor.cpp
Processing ./simulation/frc_gazebo_plugins/dc_motor/src/dc_motor.h
Processing ./simulation/frc_gazebo_plugins/encoder/src/encoder.cpp
Processing ./simulation/frc_gazebo_plugins/encoder/src/encoder.h
Processing ./simulation/frc_gazebo_plugins/gyro/src/gyro.cpp
Processing ./simulation/frc_gazebo_plugins/gyro/src/gyro.h
Processing ./simulation/frc_gazebo_plugins/limit_switch/src/switch.h
Processing ./simulation/frc_gazebo_plugins/limit_switch/src/external_limit_switch.cpp
Processing ./simulation/frc_gazebo_plugins/limit_switch/src/external_limit_switch.h
Processing ./simulation/frc_gazebo_plugins/limit_switch/src/internal_limit_switch.cpp
Processing ./simulation/frc_gazebo_plugins/limit_switch/src/internal_limit_switch.h
Processing ./simulation/frc_gazebo_plugins/limit_switch/src/limit_switch.cpp
Processing ./simulation/frc_gazebo_plugins/limit_switch/src/limit_switch.h
Processing ./simulation/frc_gazebo_plugins/pneumatic_piston/src/pneumatic_piston.cpp
Processing ./simulation/frc_gazebo_plugins/pneumatic_piston/src/pneumatic_piston.h
Processing ./simulation/frc_gazebo_plugins/potentiometer/src/potentiometer.cpp
Processing ./simulation/frc_gazebo_plugins/potentiometer/src/potentiometer.h
Processing ./simulation/frc_gazebo_plugins/rangefinder/src/rangefinder.cpp
Processing ./simulation/frc_gazebo_plugins/rangefinder/src/rangefinder.h
Processing ./simulation/frc_gazebo_plugins/servo/src/servo.cpp
Processing ./simulation/frc_gazebo_plugins/servo/src/servo.h
Processing ./wpilibc/athena/include/AnalogGyro.h
Processing ./wpilibc/athena/include/CounterBase.h
Processing ./wpilibc/athena/include/DigitalGlitchFilter.h
Processing ./wpilibc/athena/include/DigitalSource.h
Processing ./wpilibc/athena/include/I2C.h
Processing ./wpilibc/athena/include/IterativeRobot.h
Processing ./wpilibc/athena/include/Jaguar.h
Processing ./wpilibc/athena/include/Notifier.h
Processing ./wpilibc/athena/include/SD540.h
Processing ./wpilibc/athena/include/SampleRobot.h
Processing ./wpilibc/athena/include/Spark.h
Processing ./wpilibc/athena/include/Talon.h
Processing ./wpilibc/athena/include/TalonSRX.h
Processing ./wpilibc/athena/include/Victor.h
Processing ./wpilibc/athena/include/VictorSP.h
Processing ./wpilibc/athena/include/MotorSafety.h
Processing ./wpilibc/athena/include/RobotBase.h
Processing ./wpilibc/athena/include/PWMSpeedController.h
Processing ./wpilibc/athena/include/CANSpeedController.h
Processing ./wpilibc/athena/include/ADXL345_I2C.h
Processing ./wpilibc/athena/include/ADXL345_SPI.h
Processing ./wpilibc/athena/include/ADXL362.h
Processing ./wpilibc/athena/include/ADXRS450_Gyro.h
Processing ./wpilibc/athena/include/AnalogAccelerometer.h
Processing ./wpilibc/athena/include/AnalogInput.h
Processing ./wpilibc/athena/include/AnalogOutput.h
Processing ./wpilibc/athena/include/AnalogPotentiometer.h
Processing ./wpilibc/athena/include/AnalogTrigger.h
Processing ./wpilibc/athena/include/AnalogTriggerOutput.h
Processing ./wpilibc/athena/include/BuiltInAccelerometer.h
Processing ./wpilibc/athena/include/CANJaguar.h
Processing ./wpilibc/athena/include/CANTalon.h
Processing ./wpilibc/athena/include/CameraServer.h
Processing ./wpilibc/athena/include/Compressor.h
Processing ./wpilibc/athena/include/ControllerPower.h
Processing ./wpilibc/athena/include/Counter.h
Processing ./wpilibc/athena/include/DigitalInput.h
Processing ./wpilibc/athena/include/DigitalOutput.h
Processing ./wpilibc/athena/include/DoubleSolenoid.h
Processing ./wpilibc/athena/include/DriverStation.h
Processing ./wpilibc/athena/include/Encoder.h
Processing ./wpilibc/athena/include/GearTooth.h
Processing ./wpilibc/athena/include/InterruptableSensorBase.h
Processing ./wpilibc/athena/include/Joystick.h
Processing ./wpilibc/athena/include/MotorSafetyHelper.h
Processing ./wpilibc/athena/include/PWM.h
Processing ./wpilibc/athena/include/PowerDistributionPanel.h
Processing ./wpilibc/athena/include/Preferences.h
Processing ./wpilibc/athena/include/Relay.h
Processing ./wpilibc/athena/include/RobotDrive.h
Processing ./wpilibc/athena/include/SPI.h
Processing ./wpilibc/athena/include/SafePWM.h
Processing ./wpilibc/athena/include/SerialPort.h
Processing ./wpilibc/athena/include/Servo.h
Processing ./wpilibc/athena/include/Solenoid.h
Processing ./wpilibc/athena/include/SolenoidBase.h
Processing ./wpilibc/athena/include/SpeedController.h
Processing ./wpilibc/athena/include/USBCamera.h
Processing ./wpilibc/athena/include/Ultrasonic.h
Processing ./wpilibc/athena/include/WPILib.h
Processing ./wpilibc/athena/include/Internal/HardwareHLReporting.h
Processing ./wpilibc/athena/include/Vision/BaeUtilities.h
Processing ./wpilibc/athena/include/Vision/ColorImage.h
Processing ./wpilibc/athena/include/Vision/FrcError.h
Processing ./wpilibc/athena/include/Vision/HSLImage.h
Processing ./wpilibc/athena/include/Vision/RGBImage.h
Processing ./wpilibc/athena/include/Vision/Threshold.h
Processing ./wpilibc/athena/include/Vision/VisionAPI.h
Processing ./wpilibc/athena/include/Vision/AxisCamera.h
Processing ./wpilibc/athena/include/Vision/BinaryImage.h
Processing ./wpilibc/athena/include/Vision/ImageBase.h
Processing ./wpilibc/athena/include/Vision/MonoImage.h
Processing ./wpilibc/athena/src/Compressor.cpp
Processing ./wpilibc/athena/src/I2C.cpp
Processing ./wpilibc/athena/src/Joystick.cpp
Processing ./wpilibc/athena/src/Notifier.cpp
Processing ./wpilibc/athena/src/Task.cpp
Processing ./wpilibc/athena/src/USBCamera.cpp
Processing ./wpilibc/athena/src/ADXRS450_Gyro.cpp
Processing ./wpilibc/athena/src/GearTooth.cpp
Processing ./wpilibc/athena/src/Preferences.cpp
Processing ./wpilibc/athena/src/SampleRobot.cpp
Processing ./wpilibc/athena/src/MotorSafetyHelper.cpp
Processing ./wpilibc/athena/src/Relay.cpp
Processing ./wpilibc/athena/src/SafePWM.cpp
Processing ./wpilibc/athena/src/AnalogPotentiometer.cpp
Processing ./wpilibc/athena/src/BuiltInAccelerometer.cpp
Processing ./wpilibc/athena/src/PWM.cpp
Processing ./wpilibc/athena/src/PowerDistributionPanel.cpp
Processing ./wpilibc/athena/src/Servo.cpp
Processing ./wpilibc/athena/src/Ultrasonic.cpp
Processing ./wpilibc/athena/src/PWMSpeedController.cpp
Processing ./wpilibc/athena/src/ADXL345_I2C.cpp
Processing ./wpilibc/athena/src/ADXL345_SPI.cpp
Processing ./wpilibc/athena/src/ADXL362.cpp
Processing ./wpilibc/athena/src/AnalogAccelerometer.cpp
Processing ./wpilibc/athena/src/AnalogGyro.cpp
Processing ./wpilibc/athena/src/AnalogInput.cpp
Processing ./wpilibc/athena/src/AnalogOutput.cpp
Processing ./wpilibc/athena/src/AnalogTrigger.cpp
Processing ./wpilibc/athena/src/AnalogTriggerOutput.cpp
Processing ./wpilibc/athena/src/CANJaguar.cpp
Processing ./wpilibc/athena/src/CANTalon.cpp
Processing ./wpilibc/athena/src/CameraServer.cpp
Processing ./wpilibc/athena/src/ControllerPower.cpp
Processing ./wpilibc/athena/src/Counter.cpp
Processing ./wpilibc/athena/src/DigitalGlitchFilter.cpp
Processing ./wpilibc/athena/src/DigitalInput.cpp
Processing ./wpilibc/athena/src/DigitalOutput.cpp
Processing ./wpilibc/athena/src/DoubleSolenoid.cpp
Processing ./wpilibc/athena/src/DriverStation.cpp
Processing ./wpilibc/athena/src/Encoder.cpp
Processing ./wpilibc/athena/src/InterruptableSensorBase.cpp
Processing ./wpilibc/athena/src/IterativeRobot.cpp
Processing ./wpilibc/athena/src/Jaguar.cpp
Processing ./wpilibc/athena/src/PIDController.cpp
Processing ./wpilibc/athena/src/RobotBase.cpp
Processing ./wpilibc/athena/src/RobotDrive.cpp
Processing ./wpilibc/athena/src/SD540.cpp
Processing ./wpilibc/athena/src/SPI.cpp
Processing ./wpilibc/athena/src/SensorBase.cpp
Processing ./wpilibc/athena/src/SerialPort.cpp
Processing ./wpilibc/athena/src/Solenoid.cpp
Processing ./wpilibc/athena/src/SolenoidBase.cpp
Processing ./wpilibc/athena/src/Spark.cpp
Processing ./wpilibc/athena/src/Talon.cpp
Processing ./wpilibc/athena/src/TalonSRX.cpp
Processing ./wpilibc/athena/src/Timer.cpp
Processing ./wpilibc/athena/src/Utility.cpp
Processing ./wpilibc/athena/src/Victor.cpp
Processing ./wpilibc/athena/src/VictorSP.cpp
Processing ./wpilibc/athena/src/Internal/HardwareHLReporting.cpp
Processing ./wpilibc/athena/src/Vision/VisionAPI.cpp
Traceback (most recent call last):
  File "./license-update.py", line 46, in <module>
    line = file.readline()
  File "/usr/lib/python3.5/codecs.py", line 321, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x97 in position 6694: invalid start byte
`